homeassistant/components/google_generative_ai_conversation/__init__.py

```diff
@@ -130,109 +110,15 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
             return False
         raise ConfigEntryNotReady(err) from err
 
-    conversation.async_set_agent(hass, entry, GoogleGenerativeAIAgent(hass, entry))
+    await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
+
     return True
 
 
 async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Unload GoogleGenerativeAI."""
+    if not await hass.config_entries.async_unload_platforms(entry, PLATFORMS):
+        return False
+
     genai.configure(api_key=None)
-    conversation.async_unset_agent(hass, entry)
```

Comment:
FYI OpenAI PR didn't remove this and I did here since it seems like an omission. I was going to remove from OpenAI in a followup.